server/server.go

```diff
@@ -222,47 +222,46 @@ func (s *Server) handleGetMh(w lookupResponseWriter, r *http.Request) {
 	log := logger.With("multihash", mh)
 	ctx := context.Background()
 
+	// create result and error channels
 	resChan := make(chan model.ProviderResult)
-	errChan := make(chan error)
+	// the size of two is required so that the client doesn't in its go routine when trying to submit an error and return
+	errChan := make(chan error, 2)
```

Comment:
I do not see a reason that this needs to be more than 1 [here](https://github.com/ipni/go-libipni/blob/main/find/client/http/dhash_client.go#L95).  Also, if a particular channel size is needed, then the called function should return the channel that is appropriately sized. It is not reasonable to expect the caller to know that any particular size is needed.

Reply:
The reason it's two is because if it's one then it'd deadlock. The resChan gets closed in `defer`, so the client go routine hangs indefinitely on `errChan <- err`, while the calling go routine hangs indefinitely in the `for res <- resChan`. But anyway - it's not relevant if the error gets returned from the function as you suggested.